sqlmesh/core/plan/definition.py

```diff
@@ -253,14 +257,26 @@ def environment(self) -> Environment:
             if self.is_dev and self.environment_ttl is not None
             else None
         )
+
+        snapshots = [s.table_info for s in self.snapshots]
+        promoted_snapshot_ids = None
+        if self.is_dev and not self.promote_all:
+            names_to_promote = {
+                *self.context_diff.promoted_model_names,
+                *self.context_diff.added,
+                *self.context_diff.modified_snapshots,
+            } - self.context_diff.removed
```

Comment:
Want to make this into a property on context_diff? `promotable_snapshots` or something?